jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java

```diff
@@ -14,13 +19,7 @@
  */
 public class HeaderFields implements Map<String, List<String>> {
 
-    private final TreeMap<String, List<String>> content = new TreeMap<>(new Comparator<String>() {
-
-        @Override
-        public int compare(String lhs, String rhs) {
-            return lhs.compareToIgnoreCase(rhs);
-        }
-    });
+    private final ConcurrentSkipListMap<String, List<String>> content = new ConcurrentSkipListMap<>(String::compareToIgnoreCase);
```

Comment:
Won't this kill performance for high qps applications ?

Reply:
How many headers are there?

Reply:
We could synchronize the methods instead, I guess.

Reply:
Can't use a `syncrhonizedMap`, as this class implements `Map` itself, and exposes all views of the underlying map.

Reply:
Ah, there are unsynchronized `ArrayList` objects in the map, so need to do something about them as well.